### DynamicST introduction

---

DynamicST is a command-line tool developed by DynamicBiosystems for processing spatial transcriptomics data generated by the DynaSpatial system. It enables researchers to convert raw image data and sequencing data into valuable gene expression information with spatial context.

The key features of DynamicST include:

#### 1.Preprocessing and Quality Control: 
DynamicST converts raw images and fastq files into high-quality digital expression matrices. It performs necessary quality control steps to ensure accurate downstream analysis.

#### 2.Data Visualization: 
The software provides intuitive visualization results, including UMI distribution plots, clustering plots, and saturation curve plots.

#### 3.Integration with Other Tools: 
DynamicST is compatible with popular bioinformatics workflows, including Seurat and scanpy, allowing users to perform further analysis of DynaSpatial data.

DynamicST boasts powerful analytical capabilities, empowering scientists to extract insights from complex DynaSpatial gene expression data and advancing our understanding of tissue architecture, cellular microenvironment interactions, and disease biology. Its applications span across multiple disciplines in the life sciences, contributing to fields such as cancer research, neuroscience, and developmental biology.

### Install DynamicST 1.0.5

----

Dowload DynamicST from [here](https://github.com/DynamicBiosystems/DynamicST/releases/tag/v1.0.5).

```shell
# download dynamicst_v1.0.5.tar.gz

mkdir DynamicST

tar -zxf dynamicst_v1.0.5.tar.gz -C DynamicST

Prepend the DynamicST/dynamicst_v1.0.5 directory to your $PATH. This will allow you to invoke the DynamicST command.
```

### Manual

---

#### params

---

#### mkref

- `--forceall`: Force the execution of the selected (or the first) rule and all rules it is dependent on regardless of already created output.
- `--dryrun`: Do not execute anything, and display what would be done. If you have a very large workflow, use `--dryrun --quiet` to just print a summary of the DAG of jobs.
- `--genomeName`: Output dir, used to build genome.
- `--fasta`: Path to FASTA file containing your genome reference.
- `--gtf`: Path to genes GTF file containing annotated genes for your genome reference.
- `--threads`: Number of threads used during STAR genome index generation. Defaults to 8.

#### count

- `--sample`: Sample name of each batch, required.
- `--id`: Final sample name, required.
- `--inputdir`: Raw data path, required.
- `--image`: Single H&E brightfield image.
- `--dynaimage`: Brightfield image generated by the dynablot instrument (the dynablot image).
- `--alignment`: Alignment file produced by the manual alignment step.
- `--probe-set`: CSV file specifying the probe set used, if any.
- `--chip-type`: Chip type for automatic tissue detection (10mm, 5mm, 7.5mm, HD-3mm), not supported - do not use.
- `--r2-length`: Hard trim the input Read 2 to this length before analysis; default: 91.
- `--gtf`: Genome annotation file, required.
- `--transcriptome`: Path of folder containing transcriptome reference, required.
- `--outputdir`: Output file name, default: `./`.
- `--CBposition`: Position of Cell Barcode(s) on the barcode read (0-base). Default: `0_0_0_7 0_38_0_45`.
- `--UMIposition`: Position of the UMI on the barcode read, same as CBposition (0-base). Default: `0_46_0_57`.
- `--cores`: Set max cores the pipeline may request at one time; default: 16.
#### Quick start

---

- mkref

```shell
wget ftp://ftp.ensembl.org/pub/release-99/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
wget ftp://ftp.ensembl.org/pub/release-99/gtf/homo_sapiens/Homo_sapiens.GRCh38.99.gtf.gz

gunzip Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
gunzip Homo_sapiens.GRCh38.99.gtf.gz

DynamicST mkref \
 --genome_name Homo_sapiens_GRCh38 \
 --fasta Homo_sapiens.GRCh38.dna.primary_assembly.fa \
 --gtf Homo_sapiens.GRCh38.99.gtf
```

- count

```shell
#The input folder must contain fastq files, with the same file format as sampleName_S1_L001_R1_001.fastq.gz
DynamicST count  \
--sample sampleName \
--id sampleName \
--inputdir rawdata \
--gtf Homo_sapiens.GRCh38.99.gtf \
--transcriptome Homo_sapiens_GRCh38 \
--image HE.tif \
--alignment alignment.json \
--outputdir result
```

The detailed documentation of the web_summary.html in the results can be found at [summary](https://github.com/DynamicBiosystems/DynamicST/blob/main/doc/web_summary.md).

### About

---

DynamicST is developed by dynamic-biosystems Co., Ltd. The official website is [www.dynamic-biosystems](http://www.dynamic-biosystems.com/).





